src/components/Layout/RightPaneLayout/RightPaneContext.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { ReactElement, ReactNode, useContext, useMemo, useState } from "react";
+
+interface OpenRightPaneOpts {
+  content: ReactNode;
+}
+
+export type RightPaneLayoutContextProps = {
+  openInPane: (opts: OpenRightPaneOpts) => void;
+  closePane: () => void;
+  isRightPaneOpen: Boolean;
+  rightPaneContent: ReactNode;
+};
+
+export const RightPaneContext = React.createContext<RightPaneLayoutContextProps>({
+  openInPane: () => {},
+  closePane: () => {},
+  isRightPaneOpen: false,
+  rightPaneContent: null,
+});
+
+export function RightPaneProvider({ children }: { children: ReactElement }) {
+  const [rightPaneContent, setRightPaneContent] = useState<ReactNode>(undefined);
+  const [isRightPaneOpen, setIsRightPaneOpen] = useState<Boolean>(false);
+
+  const context = useMemo(() => {
+    return {
+      openInPane: (opts: OpenRightPaneOpts) => {
+        setRightPaneContent(opts?.content);
+        setIsRightPaneOpen(true);
```

Comment:
**question:** Can we get away with just one state here? Like if `rightContentPane` isn't null/undefined, then considered the right pane opened?